pkg/nsm/apis/nsmconnect/nsmconnect.proto

```diff
@@ -0,0 +1,57 @@
+syntax = "proto3";
+package nsmconnect;
+
+message Label {
+    map<string ,string> selector = 1;
+}
+
+message Metadata {
+    string Name = 1;
+    string Namespace = 2;
+    Label Labels = 3;
+}
+
+enum InterfaceType {
+    KernelInterface = 0;
+    VhostUser = 1;
+    Memif = 2;
+    Other = 3;
+  }
+
+message MechanismType {
+    InterfaceType type = 1;
+  }
+
+message AffinityPreference {
+    string Predicate = 1;
+}
+
+message ConnectionRequest {
+
+    Metadata Metadata = 1;
+    
+    string NetworkServiceName = 2;
+
+    repeated MechanismType Type = 3;
```

Comment:
Note: There may optionally be Mechanism preferences.

If I'm asking for a Kernel Interface, I may want to request a particular interface name within my Pod's namespace
If I'm asking for a memif or vhostuser I may want to ask for a specific socket name (they get set up over unix file sockets) etc.  Mechanism parameters will vary by MechanismType.

Reply:
Great idea, being able to request a specific interface name could be very useful. I guess (maybe not right now) certain specific interface parameters could also be requested, like MTU, enable/disable some interface features.
WRT Mechanism Preference, how do you see it, add `Enumeration Choice` with values 1st, 2nd, 3rd etc?